apps/server/src/core/error/filter/global-error.filter.spec.ts

```diff
@@ -304,24 +323,82 @@ describe('GlobalErrorFilter', () => {
 					).toBeCalledWith(expectedResponse);
 				});
 			});
+
+			describe('when error has a cause error', () => {
+				const setup = () => {
+					const causeError = new Error('Cause error');
+					const error = new LoggableExceptionWithCause('test', causeError);
+					const expectedResponse = new ErrorResponse(
+						'WITH_CAUSE',
+						'With Cause',
+						'Loggable Exception With Cause',
+						HttpStatus.INTERNAL_SERVER_ERROR
+					);
+
+					const argumentsHost = setupHttpArgumentsHost();
+
+					return { error, argumentsHost, expectedResponse };
+				};
+
+				it('should set response status appropriately', () => {
+					const { error, argumentsHost } = setup();
+
+					service.catch(error, argumentsHost);
+
+					expect(argumentsHost.switchToHttp().getResponse<Response>().status).toBeCalledWith(
+						HttpStatus.INTERNAL_SERVER_ERROR
+					);
+				});
+
+				it('should send appropriate error response', () => {
+					const { error, argumentsHost, expectedResponse } = setup();
+
+					service.catch(error, argumentsHost);
+
+					expect(
+						argumentsHost.switchToHttp().getResponse<Response>().status(HttpStatus.INTERNAL_SERVER_ERROR).json
+					).toBeCalledWith(expectedResponse);
+				});
+			});
 		});
 
 		describe('when context is rmq', () => {
-			const setup = () => {
-				const argumentsHost = createMock<ArgumentsHost>();
-				argumentsHost.getType.mockReturnValueOnce('rmq');
+			describe('Name of the group', () => {
+				const setup = () => {
+					const argumentsHost = createMock<ArgumentsHost>();
+					argumentsHost.getType.mockReturnValueOnce('rmq');
 
-				const error = new Error();
+					const error = new Error();
 
-				return { error, argumentsHost };
-			};
+					return { error, argumentsHost };
+				};
 
-			it('should return an RpcMessage with the error', () => {
-				const { error, argumentsHost } = setup();
+				it('should return an RpcMessage with the error', () => {
+					const { error, argumentsHost } = setup();
 
-				const result = service.catch(error, argumentsHost);
+					const result = service.catch(error, argumentsHost);
 
-				expect(result).toEqual({ message: undefined, error });
+					expect(result).toEqual({ message: undefined, error });
+				});
+			});
+
+			describe('when error is a LoggableError', () => {
+				const setup = () => {
+					const causeError = new Error('Cause error');
+					const error = new LoggableExceptionWithCause('test', causeError);
+					const argumentsHost = createMock<ArgumentsHost>();
+					argumentsHost.getType.mockReturnValueOnce('rmq');
+
+					return { error, argumentsHost };
+				};
+
+				it('should return appropriate error', () => {
+					const { error, argumentsHost } = setup();
+
+					const result = service.catch(error, argumentsHost);
+
+					expect(result).toEqual({ message: undefined, error });
+				});
 			});
 		});
```

Comment:
Wäre hier noch ein describe sinnvoll wo wir ein Szenario wie "describe('when context is other than rmq and http'," testen?